src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferLengthCountIsEmptyOverAnyAnalyzer.cs

```diff
@@ -0,0 +1,155 @@
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false
+        );
+
+        internal static readonly DiagnosticDescriptor CountDescriptor = DiagnosticDescriptorHelper.Create(
+            CountId,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false
+        );
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
+            LengthDescriptor,
+            CountDescriptor,
+            IsEmptyDescriptor
+        );
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterOperationAction(OnInvocationAnalysis, OperationKind.Invocation);
+        }
+
+        private static void OnInvocationAnalysis(OperationAnalysisContext context)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+            var originalMethod = invocation.TargetMethod.OriginalDefinition;
+            if (originalMethod.MethodKind == MethodKind.ReducedExtension)
+            {
+                originalMethod = originalMethod.ReducedFrom;
+            }
+
+            if (IsEligibleAnyMethod(originalMethod))
+            {
+                var type = invocation.GetReceiverType(context.Compilation, beforeConversion: true, context.CancellationToken);
+                if (type is null)
+                {
+                    return;
+                }
+
+                if (HasEligibleIsEmptyProperty(type))
+                {
+                    context.ReportDiagnostic(invocation.CreateDiagnostic(IsEmptyDescriptor));
+
+                    return;
+                }
+
+                if (HasEligibleLengthProperty(type))
+                {
+                    context.ReportDiagnostic(invocation.CreateDiagnostic(LengthDescriptor));
+
+                    return;
+                }
+
+                if (HasEligibleCountProperty(type))
+                {
+                    context.ReportDiagnostic(invocation.CreateDiagnostic(CountDescriptor));
+
+                    return;
+                }
+            }
+        }
+
+        private static bool IsEligibleAnyMethod(IMethodSymbol method)
+        {
+            return method is
+            {
+                Name: AnyText,
```

Comment:
This diagnostic is about _specific_ implementations of `Count()` and `Any()`. We only know this is true for the one that come from the framework. If there are other extension methods that _happen_ to have these names we would have no idea if this is a correct thing to suggest. I would like us to also match on the namespace that the extension methods come from to reduce the number of false positives.

Reply:
Yes, this should specifically be about System.Linq.Enumerable.Count and System.Linq.Enumerable.Any.

Reply:
I had [similar comment about this](https://github.com/dotnet/roslyn-analyzers/pull/6236#discussion_r1063840526) and had offline discussion and my understanding was we prefer to apply the rule for Any() method of other types too. Though personally I prefer to restrict it to `Enumerable.Any`. 

@CollinAlpert please [do a symbol check](https://github.com/dotnet/roslyn-analyzers/pull/6236#discussion_r1063703944) instead as @Youssef1313 suggested before

Reply:
> my understanding was we prefer to apply the rule for Any() method of other types too

If we know what those methods are because we own them, I think that's fine, e.g. you had asked me offline about ImmutableArrayExtensions.Any, which is an extension defined in System.Collections.Immutable, i.e. we own it.  I'm fine if we want to special-case that in addition to System.Linq.Enumerable.Any.  But the biggest impact here and the simplest thing to do is focus on just LINQ's Enumerable.Any.

Reply:
If there are multiple `Any` methods that are defined in dotnet/runtime that we want to check for, I think we should check for those specific methods. I don't expect the list of `Any` extension methods in the framework will be large enough for the list to be hard to maintain.

Reply:
> I think that's fine, e.g. you had asked me offline about ImmutableArrayExtensions.Any, which is an extension defined in System.Collections.Immutable, i.e. we own it. 

Sorry, I misunderstood, I took `ImmutableArrayExtensions.Any` as an example for any other `Any()` implementation 

> I'm fine if we want to special-case that in addition to System.Linq.Enumerable.Any. But the biggest impact here and the simplest thing to do is focus on just LINQ's Enumerable.Any.

Right, for me we should focus on just `Enumerable.Any`, some people might argue that `ImmutableArrayExtensions.Any()` is a false positive.